File: src/main/java/com/jme/asset/store/db/entity/asset/FileEntity.java
package com.jme.asset.store.db.entity.asset;

import com.jme.asset.store.db.entity.BaseEntity;
import com.jme.asset.store.db.entity.user.UserEntity;
import org.jetbrains.annotations.NotNull;

import javax.persistence.*;
import java.sql.Blob;

/**
 * The File entity
 *
 * @author Gomanchuk Mikhail
 */
@Entity
@Table(name = "FILE_ENTITY")
public class FileEntity extends BaseEntity {

    /**
     * The file name
     */
    @Column(name = "name", nullable = false)
    private String name;

    /**
     * The file content
     */
    @Column(name = "content", nullable = false)
    private Blob content;

    /**
     * The creator of file
     */
    @OneToOne
    @JoinColumn(name = "creator_id")
    private UserEntity creator;

    /**
     * Type of file
     */
    @OneToOne
    @JoinColumn(name = "type_id")
    private FileTypeEntity type;

    public FileEntity() {
    }

    public FileEntity(final long id, @NotNull final String name, @NotNull final Blob content,
                      @NotNull final UserEntity creator, @NotNull final FileTypeEntity type) {
        super(id);
        this.name = name;
        this.content = content;
        this.creator = creator;
        this.type = type;
    }

    /**
     * Get the file name
     *
     * @return the file name
     */
    public @NotNull String getName() {
        return name;
    }

    /**
     * Set the file name
     *
     * @param name the file name
     */
    public void setName(@NotNull final String name) {
        this.name = name;
    }

    /**
     * Get the file content
     *
     * @return the content of file
     */
    public @NotNull Blob getContent() {
        return content;
    }

    /**
     * Set the file content
     *
     * @param content the content of file
     */
    public void setContent(@NotNull final Blob content) {
        this.content = content;
    }

    /**
     * Get the file creator
     *
     * @return the file creator
     */
    public @NotNull UserEntity getCreator() {
        return creator;
    }

    /**
     * Set the file creator
     *
     * @param creator the file creator
     */
    public void setCreator(@NotNull final UserEntity creator) {
        this.creator = creator;
    }

    /**
     * Get file's type
     *
     * @return type
     */
    public @NotNull FileTypeEntity getType() {
        return type;
    }

    /**
     * Set file's type
     *
     * @param type file's type
     */
    public void setType(@NotNull final FileTypeEntity type) {
        this.type = type;
    }
}

File: front-end/src/model/entity/file-entity.ts
export class FileEntity {

  private _id: number;
  private _content: Blob;
  private _name: string;
  private _creator_id: number;
  private _type_id: number;

  constructor(id: number, content: Blob, name: string, creator_id: number, type_id: number) {
    this._id = id;
    this._content = content;
    this._name = name;
    this._creator_id = creator_id;
    this._type_id = type_id;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get content(): Blob {
    return this._content;
  }

  set content(value: Blob) {
    this._content = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get creator_id(): number {
    return this._creator_id;
  }

  set creator_id(value: number) {
    this._creator_id = value;
  }

  get type_id(): number {
    return this._type_id;
  }

  set type_id(value: number) {
    this._type_id = value;
  }
}
